### PR TITLE
munam/LP-bootcamps

### PR DESCRIPTION
**Pull request by MunamMubashir-arbisoft**
[Ticket document for this PR](https://docs.google.com/document/u/1/d/1HxSVeUHEYa4J7N2qUEDxuEe6q_U09Nfjghr85aizIx8/edit)

- Create the Bootcamps application and all it's corresponding functionality.
- Write tests for the Bootcamps application.

[PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/27) for application templates in `adg-edx-theme` directory.

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/445/workflows/workflows_id/jobs/445/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/common/lib/mandrill_client/client.py (100%)
openedx/adg/lms/bootcamps/admin.py (0.0%): Missing lines 4,6,9-10,14-15
openedx/adg/lms/bootcamps/apps.py (0.0%): Missing lines 4,7,11
openedx/adg/lms/bootcamps/models.py (0.0%): Missing lines 4-5,7,10,14-15,20-22,24-25
openedx/adg/lms/bootcamps/templatetags/bootcamp_extras.py (92.3%): Missing lines 36
openedx/adg/lms/bootcamps/templatetags/constants.py (100%)
openedx/adg/lms/bootcamps/templatetags/helpers.py (100%)
openedx/adg/lms/bootcamps/tests/test_bootcamp_extras.py (100%)
openedx/adg/lms/bootcamps/tests/test_helpers.py (100%)
openedx/adg/lms/bootcamps/urls.py (0.0%): Missing lines 4,6,8-9
openedx/adg/lms/bootcamps/views.py (0.0%): Missing lines 4,6,9,13-15,18,22-24
openedx/adg/lms/student/helpers.py (39.5%): Missing lines 28-30,35,40,54-55,64,69,84,90,94,108,122-124,129,133,147-150,152-153,168-169
openedx/adg/lms/utils/env_utils.py (66.7%): Missing lines 12
-------------
Total:   172 lines
Missing: 62 lines
Coverage: 63%
-------------
```

File: openedx/adg/lms/bootcamps/urls.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+All urls for Bootcamps application.
+"""
+from django.urls import path
+
+from .views import BootcampDetailView, BootcampsListView
+
+app_name = "bootcamps"
+urlpatterns = [
+    path('', BootcampsListView.as_view(), name="bootcamp_list"),
+    path('<int:pk>/', BootcampDetailView.as_view(), name="bootcamp_detail"),
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/urls.py:11]: single qoutes please

File: openedx/adg/lms/bootcamps/views.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+from .models import Bootcamp
+
+
+class BootcampsListView(ListView):
+    """
+    Generic view to display the list of all Bootcamps
+    """
+    model = Bootcamp
+    context_object_name = "bootcamps"
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/views.py:14]: single qoutes here and in line 23

File: openedx/adg/lms/bootcamps/tests/test_bootcamp_extras.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+"""
+Tests for all the code in templatetags/bootcamp_extras.py in Bootcamps application.
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/tests/test_bootcamp_extras.py:2]: a little redundant, maybe change it to:

"Tests for template tags"

File: openedx/adg/lms/bootcamps/tests/test_bootcamp_extras.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class BootcampExtrasTestCases(ModuleStoreTestCase):
+    """
+    TestCase for the bootcamp_extras functions
+    """
+    USERNAME = "test"
+    EMAIL = "test@example.com"
+    PASSWORD = "edx"
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/tests/test_bootcamp_extras.py:25]: single qoutes

File: openedx/adg/lms/bootcamps/tests/test_helpers.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+"""
+Tests for all the code in templatetags/helpers.py Bootcamps application.
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/tests/test_helpers.py:2]: a little redundant, maybe change it to:

"Tests for template tags' helper methods"

File: openedx/adg/lms/bootcamps/tests/test_helpers.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class HelpersTestCase(ModuleStoreTestCase):
+    """
+    TestCase class for the helpers.py functions
+    """
+    USERNAME = "test"
+    EMAIL = "test@example.com"
+    PASSWORD = "edx"
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/tests/test_helpers.py:26]: the same data is created in `test_bootcamp_extras.py`. Let's move it to a `constants.py`.

**Review by harris-irfan — CHANGES_REQUESTED** [review]
- Please commit and push missing files in the PR: `lms/envs/common.py` where the new application is registered and the file used to add application url `openedx/adg/lms/urls.py`.
- Please provide the link of the ticket document in the PR description
- Please provide the link of the `adg-edx-theme` PR in the PR description
- Please have a look at the suggested changes

File: openedx/adg/lms/bootcamps/templatetags/constants.py [hunk 7]
```diff
@@ -2,8 +2,10 @@
...
-IN_PROGRESS = 1
-COMPLETED = 2
+NOT_STARTED = 'Not Started'
+IN_PROGRESS = 'In Progress'
+COMPLETED = 'Completed'
 
-MAP_PROGRESS_INTEGER_TO_STRING = ["Not Started", "In Progress", "Completed"]
+USERNAME = 'test'
+EMAIL = 'test@example.com'
+PASSWORD = 'edx'
```
Comment (harris-irfan) [inline openedx/adg/lms/bootcamps/templatetags/constants.py:11]: Are these constants used anywhere in templatetags? If they are solely used in the tests, then a new file for constants should be placed in the /tests directory, and they should be placed in that file.

**Review by harris-irfan — CHANGES_REQUESTED** [review]

**Comment by harris-irfan** [comment]
Please also commit any new files that you have added, like adg_utils.

File: lms/envs/common.py [hunk 8]
```diff
@@ -44,17 +43,15 @@
...
 from openedx.core.djangoapps.theming.helpers_dirs import (
     get_themes_unchecked,
     get_theme_base_dirs_from_settings
 )
 from openedx.core.lib.derived import derived, derived_collection_entry
 from openedx.core.release import doc_version
-from xmodule.modulestore.modulestore_settings import update_module_store_settings
+from path import Path as path
 from xmodule.modulestore.edit_info import EditInfoMixin
-from lms.djangoapps.lms_xblock.mixin import LmsBlockMixin
```
Comment (harris-irfan) [inline lms/envs/common.py:57]: we usually avoid making changes in core files like these, even when it comes to isorting imports. Let's only make changes that are necessary in core files

Comment (MunamMubashir-arbisoft) [reply lms/envs/common.py:57]: Makes sense. My bad.

**Review by harris-irfan — APPROVED** [review]
LGTM, Well done!

**Pull request closed without merging**